### DISCUSSION TITLE
Consider use of http-streams

### DISCUSSION THREAD

**Discussion by istathar** [aesiniath/unbeliever#160] [Internals]
I like **http-streams**, because I wrote it! and because I like the **io-streams** library's approach to the Streaming I/O problem.

It was, however, the first Haskell library I wrote, and, though _not_ unmaintained I have hardly touched it in years. It works _fine_, as evidenced by the large amount of traffic that is driven using it at a number of companies.

But there's no one who is really looking after it, and I'm a bit worried about some of the things in its dependency stack in turn bitrotting.

Inside this open source project it is used by **core-telemetry** as the HTTP client library which connects to Honeycomb and sends events. We _could_ consider replacing it with **http-client** (which would be ironic, seeing as how I first wrote **http-streams** because I didn't like the semantics of Snoyman's Conduit based client) as it is the defacto standard implementation in Haskell and more likely to be better tested around narly edge cases.

Or, we could invest in ensuring **http-streams** is up to date and possibly bring it under the umbrella of this project.

Thoughts?

**Comment by istathar** [comment]
The relevant code in **core-telemetry** is https://github.com/aesiniath/unbeliever/blob/e67a84f719d8fc0d6d5114e36f95cf63aaec0744/core-telemetry/lib/Core/Telemetry/Honeycomb.hs#L216-L292

**Comment by istathar** [comment]
(an equally helpful outcome of this discussion would be "it's fine!" but I'd love someone else to weigh in)

**Reply by jaesharp** [reply]
Taking a look at this, but I want to load up some context so it may take me a little bit to get back to you.
